block-producer/src/server/api.rs

```diff
@@ -38,9 +40,14 @@ where
     ) -> Result<tonic::Response<SubmitProvenTransactionResponse>, Status> {
         let request = request.into_inner();
 
+        let request_id = gen_request_id();
+        let _span = info_span!("submit_proven_transaction", request_id, ?request, COMPONENT);
+
         let tx = ProvenTransaction::read_from_bytes(&request.transaction)
             .map_err(|_| Status::invalid_argument("Invalid transaction"))?;
 
+        debug!(request_id, ?tx, COMPONENT);
```

Comment:
Similar to the last comment, I think this may be too verbose as `ProvenTransactions` may be dozens of KBs of binary data. Ideally, we'd log the following:

1. At `info` level something like: "parsed proven transaction" with transaction ID.
2. At `debug` level all details of the transaction except for the `proof`.

For the second point, I'd probably create a wrapper struct with `Display` implementation - though maybe doing it field-by-field could work too.

Reply:
@bobbinth got it!